seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/FieldMapperTransform.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.transform;
+
+import org.apache.seatunnel.api.configuration.Option;
+import org.apache.seatunnel.api.configuration.Options;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.api.transform.SeaTunnelTransform;
+import org.apache.seatunnel.common.utils.JsonUtils;
+import org.apache.seatunnel.transform.common.AbstractSeaTunnelTransform;
+import org.apache.seatunnel.transform.exception.FieldMapperTransformErrorCode;
+import org.apache.seatunnel.transform.exception.FieldMapperTransformException;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigRenderOptions;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.auto.service.AutoService;
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@AutoService(SeaTunnelTransform.class)
+public class FieldMapperTransform extends AbstractSeaTunnelTransform {
+    public static final Option<Map<String, String>> FIELD_MAPPER = Options.key("field_mapper")
+        .mapType()
+        .noDefaultValue()
+        .withDescription("Specify the field mapping relationship between input and output");
+
+    private LinkedHashMap<String, String> fieldMapper = new LinkedHashMap<>();
+
+    private List<Integer> needReaderColIndex;
+
+    @Override
+    public String getPluginName() {
+        return "FieldMapper";
+    }
+
+    @Override
+    protected void setConfig(Config pluginConfig) {
+        if (!pluginConfig.hasPath(FIELD_MAPPER.key())) {
+            throw new IllegalArgumentException("The configuration missing key: " + FIELD_MAPPER);
+        }
+        this.fieldMapper = convertConfigToMap(pluginConfig.getConfig(FIELD_MAPPER.key()));
+    }
+
+    private static LinkedHashMap<String, String> convertConfigToMap(Config config) {
+        // Because the entrySet in typesafe config couldn't keep key-value order
+        // So use jackson parsing schema information into a map to keep key-value order
+        ConfigRenderOptions options = ConfigRenderOptions.concise();
+        String mapper = config.root().render(options);
+        return convertJsonToMap(mapper);
+    }
+
+    private static LinkedHashMap<String, String> convertJsonToMap(String json) {
```

Comment:
```suggestion
    private static LinkedHashMap<String, String> convertConfigToSortedMap(Config config) {
        // Because the entrySet in typesafe config couldn't keep key-value order
        // So use jackson parsing schema information into a map to keep key-value order
        ConfigRenderOptions options = ConfigRenderOptions.concise();
        String json = config.root().render(options);
```